spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -310,4 +263,62 @@ public String toString() {
         "IcebergScan(table=%s, type=%s, filters=%s, runtimeFilters=%s, caseSensitive=%s)",
         table(), expectedSchema().asStruct(), filterExpressions(), runtimeFilterExpressions, caseSensitive());
   }
+
+  static class Context {
```

Comment:
Looks like this comes entirely from `SparkReadConf`, which is also passed into `SparkBatchQueryScan`. Instead of creating the context and passing it in, why not create it in this class? But then, if it is only used in this class, why not leave `equals` and `hashCode` unchanged and not use this at all?

Reply:
Like I wrote [here](https://github.com/apache/iceberg/pull/3763#discussion_r771475824), I am not happy with this place either.

For merge-on-read scans, the snapshot ID does not come from `SparkReadConf`. Instead, it comes from the current table snapshot no matter what. `SparkScanBuilder` has more context. Let me think more about this place.

Reply:
Seems like initializing `SparkReadConf` with a specific snapshot would work then?